src/science/roughness/cable_roughness.F90

```diff
@@ -35,19 +34,100 @@ MODULE cable_roughness_module
+! water vapour fluxes) and bulk formulae for the roughness length and 
+! displacement height accounting for roughness sublayer effects (for the 
+! momentum flux).
+!
+! The calculations take into account whether:
+! 
+! 1. the soil model used is SLI or the default (soilsnow)
+! 2. the land point is vegetated (LAI > LAI_THRESH) or not
+!
+! The primary references for these formulations are
+!
+! * [Raupach MR (1989)](https://doi.org/10.1002/qj.49711548710)
+! * [Raupach MR (1992)](https://doi.org/10.1007/BF00155203)
+! * [Raupach MR (1994)](https://doi.org/10.1007/BF00709229)
+! * [Kowalczyk et al. (2006)](http://www.cmar.csiro.au/e-print/open/kowalczykea_2006a.pdf)
+!
+! <br></br>
+! The principal outputs from the MODULE are
+!
+! * `rough%zref_tq`: height above the displacement height for the 
+!    air temperature and humidity (m)
+! * `rough%zref_uv`: height above the displacement height for the 
+!    wind speeds (m)
+! * `rough%hruff`: the canopy height accounting for the presence 
+!    of snow (m)
+! * `canopy%vlaiw`: the leaf area index accounting for the presence
+!    of snow (\(m^2/m^2\))
+! * `rough%z0soil`: the aerodynamic roughness length for soil (m)
+! * `rough%z0ssoilsn`: the aerodynamic roughness lenght for snow (m)
+! * `roug%z0m`: the aerodynamic roughness length for the surface 
+!    (canopy+soil+snow) (m)
+! * `rough%disp`: the displacement height of the surface 
+!    (=0.0 if not vegetated) (m)
+! * `rough%zruffs`: the depth of the roughness sublayer for vegetated
+!    tiles (m)
+! * `rough%coexp`: the extinction coefficient for the wind speed profile 
+!    within the canopy, \(c_{0}\) (m). \(c_{0}\) is defined to be the 
+!    coefficient within an expontial wind profile, i.e.
+!    the wind speed at height \(z\) above the ground within a canopy of 
+!    height \(h_c\) is given by \( U(z) = U_{h} \exp\{ c_{0} (z-h_c) \} \) 
+!    where \(U_h\) is the wind speed at canopy top.
+! * `rough%usuh`: the ratio of the friction velocity, \(u_*\), to the wind 
+!    speed at canopy top (-)
+! * `rough%rt0us`: *normalised* aerodynamic resitance for the turbulent
+!    transfer from the soil/snow surface to the displacement height (-)
+! * `rough%rt1us`: *normalised* aerodynamic resistance for the turbulent 
+!    transfer from the displacement height to the reference level (-)
+!
+! The aerodynamic resistances for the current time step are evaluated later by dividing
+! the *normalized resistances* by the current time step's friction velocity.
+! rough%rt1us is evaluated in three subparts (%rt1usa, %rt1usb, and %rt1usc).
+! Each of the normalized resistances are given by theoretical formulae as given by
+! the references.  One of the resistance terms (rough%rt1usc) is evaluated in
+! [[cable_canopy_module]].
+!
+!
+! Further detail given in SUBROUTINE [[ruff_resist]] 
+
+
 IMPLICIT NONE
 
 real, parameter :: z0soilsn_min = 1.e-7
+  !! Minimum value for the roughness length for bare soil, `(z_{0,min}`) (m)
```

Comment:
I moved the description of the variables with the declaration as I think that's the best standard to use when possible (not derived type). 

Where does the notation \(z_{0,min}\) come from? Is that a standard notation for that quantity?
```suggestion
  !! Minimum value for the roughness length for bare soil, \(z_{0,min}\) (m)
```
And I've just noted I made a typo.